google-cloud-spanner-admin-database-v1/lib/google/cloud/spanner/admin/database/v1/database_admin/client.rb

```diff
@@ -768,9 +788,25 @@ def set_iam_policy request, options = nil
                     gapic_version: ::Google::Cloud::Spanner::Admin::Database::V1::VERSION
                   metadata[:"x-goog-user-project"] = @quota_project_id if @quota_project_id
 
-                  header_params = {
-                    "resource" => request.resource
-                  }
+                  header_params = {}
+                  if request.resource
```

Comment:
The repeated `if request.resource` here could be refactored.

Reply:
I'm thinking about it but it's within the realm of 'ugliness' one can expect from the generated code and it makes the code in the generator a bit harder to navigate.
And in some cases you don't want to do it even.
But it's on my radar.